files/nrpe/check_hwclock.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+# This script will check hwclock of the node.
+# Then compare hwclock with system time.
+
+# Nagios return codes
+OK=0
+WARNING=1
+CRITICAL=2
+UNKNOWN=3
+
+HWtime=$(sudo hwclock --debug | grep Hw | cut -d" " -f 8)
+SYStime=$(date +%s)
+let diff_time=$SYStime-$HWtime
+
+if [ $diff_time -gt 300 ] ; then
+   echo "CRITICAL: Hardware clock is more than $diff_time seconds past now"
+   exit $CRITICAL
+
+elif [ $diff_time -lt 300 ] && [ $diff_time -gt 1 ] ; then
```

Comment:
These conditionals do not take into account if the difference is negative, which it would be if hwclock is in the future. What happens is that it returns "UNKNOWN" if it's just a single second in the future.

<pre>
$ sudo hwclock --set --date "2019/09/25 12:31:41"
$ bash check_hwclock.sh 
CRITICAL: Hardware clock is more than 86484 seconds past now
$ sudo hwclock --set --date "2019/09/27 12:31:41"
$ bash check_hwclock.sh 
Something Unknown happened here
</pre>

Reply:
I have also though of hwclock in future but my small knowledge says that impractical though possible. System boots with hwclock time then ntpd adjust system time with ntp servers. There is minimal chance of being hwclock in future also we have never seen that scenario. I dont know, if you think I set that way too...

Reply:
I have made the changes so that script just check the difference between hwclock and system time.

Reply:
Johans point is still valid. The check should never return UNKNOWN except in cases where the script fails. Basically if the script return UNKNOWN, it is a bug in the script.

Reply:
Yes I have considered that facts now: Hwclock is ahead of system time will be considered as Critical while Unknown is something we dont know much.

Reply:
@oscarkraemer would you consider this resolved now?

One scenario that might be justified for UNKNOWN is some weird situation where `hwclock`, `grep` or `cut` are not found in `$PATH` all of a sudden. Capturing that might require `set -e`, which we do in a handful of monitoring scripts.

Reply:
no one had any thoughts so resolving